src/WCSimRootEvent.cc

```diff
@@ -885,6 +896,7 @@ bool WCSimRootCherenkovHitTime::CompareAllVariables(const WCSimRootCherenkovHitT
   failed = (!ComparisonPassed(fTruetime, c->GetTruetime(), typeid(*this).name(), __func__, "Truetime")) || failed;
   failed = (!ComparisonPassed(fParentSavedTrackID, c->GetParentID(), typeid(*this).name(), __func__, "ParentSavedTrackID")) || failed;
   failed = (!ComparisonPassed(fPhotonStartTime, c->GetPhotonStartTime(), typeid(*this).name(), __func__, "PhotonStartTime")) || failed;
+  failed = (!ComparisonPassed(fPhotonCreatorProcess, c->GetPhotonCreatorProcess(), typeid(*this).name(), __func__, "PhotonCreatorProcess")) || failed;
```

Comment:
Please add a similar check for the WCSimRootTrack's creator process

Reply:
I do not understand this check very well, I did something like:
`failed = (!ComparisonPassed(fCreatorProcess, c->GetCreatorProcessName(), typeid(*this).name(), __func__, "CreatorProcess")) || failed;`
What do you think?

Reply:
What this does, is for a pair of `WCSimRootCherenkovHitTime`s, it checks whether they are equivalent. 
* `this` (can access data using private data members like `fPhotonCreatorProcess`)
* `c` (need to use public methods to access the data e.g. `c->GetPhotonCreatorProcess()`)

Then the rest of the line is just calling the equivalency check function `ComparisonPassed()` (the last 3 arguments used for printing out when there are failures, so we know where the failures are - the class, the function, the data member strings) & saving the result (the stuff with `failed`)

So yes, by eye it looks good. It needs to be put in the appropriate place in `WCSimRootTrack::CompareAllVariables()`

Reply:
👍